### PR TITLE
Closes #5308: Improve ArkoudaExtensionArray._reduce

### LINKED ISSUE #5308: Improve ArkoudaExtensionArray._reduce
Improve `ArkoudaExtensionArray._reduce`, add additional operators, docstring, and additional unit tests.

### PR DESCRIPTION
**Pull request by ajpotts**
## Summary

This PR modernizes the `ArkoudaArray._reduce` implementation to better align with
pandas ExtensionArray semantics, and significantly expands unit test coverage for reduction behavior.

Key goals of this change:
- Make `_reduce` string-driven, matching pandas’ calling conventions
- Support a broader, well-defined set of reductions
- Improve error reporting and dtype validation
- Add comprehensive tests covering correctness, edge cases, and failure modes

---

### ArkoudaArray._reduce
- Replaced ad-hoc conditional logic with a structured dispatch table
- Added support for:
  - Numeric reductions: `sum`, `prod`, `min`, `max`, `mean`, `var`, `std`
  - Index reductions: `argmin`, `argmax`, `first`
  - Boolean reductions: `any`, `all`, `or`, `and` (with dtype enforcement)
  - Count via array size / non-NaN count for floats
  - Mode (array-returning reduction)
- Improved error handling:
  - Unknown reduction names raise `ValueError`
  - Unsupported reductions or dtype mismatches raise `TypeError`
- Added full type hints and a darglint-compliant docstring

### GroupByReductionType
- Fixed a typo: `NUNUNIQUE` → `NUNIQUE`

### Tests
- Replaced minimal smoke tests with a dedicated `TestArkoudaArrayReduce` suite
- Added tests for:
  - Numeric reductions vs NumPy
  - Boolean reductions and dtype validation
  - `count`, `argmin`, `argmax`, and `first`
  - Empty-array edge cases
  - Unknown reduction names
  - Acceptance of the `skipna` argument
- Removed outdated tests that expected `"mean"` to be unsupported

---

## Rationale

`_reduce` is part of the pandas ExtensionArray protocol and is invoked directly
by pandas using string operation names.

The expanded test coverage ensures correctness, documents intended semantics,
and provides a stable foundation for adding future reductions (e.g., `median`,
`unique`, `nunique`) as backend support evolves.

---

## Notes for Reviewers

- This PR does **not** change public pandas-facing APIs, only internal behavior
- Reduction semantics intentionally mirror NumPy/pandas where possible
- Follow-up PRs may add capability-gated reductions as Arkouda support expands

Closes #5308: Improve ArkoudaExtensionArray._reduce

### CODE REVIEW

File: arkouda/pandas/extension/_arkouda_array.py [hunk 1]
```diff
@@ -227,21 +227,80 @@ def equals(self, other):
...
+        # All listed reductions are guaranteed to exist on pdarray for all dtypes
+        scalar_fns: dict[str, Callable[[], Any]] = {
+            "sum": data.sum,
+            "count": _count_nonmissing,
+            "prod": data.prod,
+            "min": data.min,
+            "max": data.max,
+            "mean": data.mean,
+            "var": data.var,
+            "std": data.std,
```
[the remark this answers is not in the document]
Comment (ajpotts) [reply arkouda/pandas/extension/_arkouda_array.py]: Fixed.

File: arkouda/pandas/extension/_arkouda_array.py [hunk 2]
```diff
@@ -227,21 +227,80 @@ def equals(self, other):
...
+        Parameters
+        ----------
+        name : str
+            Name of the reduction to perform (e.g., ``"sum"``, ``"min"``, ``"std"``).
+        skipna : bool
+            Whether to ignore missing values. Accepted for pandas compatibility.
+            Default is True.
+
+            Arkouda-backed arrays currently do not maintain a separate missing-value mask;
+            behavior for NaN values is determined by the underlying Arkouda operation.
```
[the remark this answers is not in the document]
Comment (ajpotts) [reply arkouda/pandas/extension/_arkouda_array.py]: I added a clearer note to the docstring pointing out that this is not fully implemented, but it is required by pandas to be in the signature.  I also created a ticket so we remember to revisit this: https://github.com/Bears-R-Us/arkouda/issues/5319
